packages/modelserver-client/src/model/diagnostic.ts

```diff
@@ -135,4 +136,82 @@ export namespace Diagnostic {
             return [];
         }
     }
+
+    /**
+     * Get an OK diagnostic.
+     *
+     * @returns an OK diagnostic
+     */
+    export function ok(): Diagnostic {
+        return {
+            severity: OK,
+            code: 0,
+            message: 'OK',
+            source: '',
+            data: [],
+            children: [],
+            id: ''
+        };
+    }
+
+    /**
+     * Merge any number of `diagnostics` (even zero) into a single result.
+     * Any OK diagnostics are elided and the result may itself just be OK.
+     *
+     * @param diagnostics diagnostics to merge
+     * @returns the merged diagnostic
+     */
+    export function merge(...diagnostics: Diagnostic[]): Diagnostic {
+        const nonOK = diagnostics.filter(d => d && d.severity > OK);
+        if (nonOK.length === 0) {
+            return ok();
+        }
+        if (nonOK.length === 1) {
+            return nonOK[0];
+        }
+
+        const max = worstOf(diagnostics);
+
+        // Recompute IDs of the diagnostics because they are now in
+        // a new tree structure and their paths have changed.
+        return recomputeIDs({
+            severity: max.severity,
+            message: `Diagnosis of ${nonOK.length} problems.`,
+            source: max.source,
+            code: max.code,
+            data: [],
+            children: nonOK,
+            id: '/'
+        });
+    }
+
+    /**
+     * Find the worst severity in a list of `diagnostics`.
+     */
+    function worstOf(diagnostics: Diagnostic[]): Diagnostic {
+        let result = diagnostics[0];
```

Comment:
Should we handle the case that the list is empty?

Reply:
We don't need to because it's an unexported function that is only called with non-empty lists. But for generality, I'm happy to make that change. And then it could even be exported?